src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -139,19 +140,33 @@ export class TerminalInstance implements ITerminalInstance {
 		this._initDimensions();
 		this._createProcess(this._shellLaunchConfig);
 		this._createXterm();
+		this.setTitle(path.basename(this._shellLaunchConfig.executable), true);
 
 		if (platform.isWindows) {
 			this._processReady.then(() => {
 				this._windowsShellHelper = new WindowsShellHelper(this._processId, this._shellLaunchConfig.executable);
+				this._onCheckWindowsShell = new Emitter<TPromise<string>>();
+				debounceEvent(this._onCheckWindowsShell.event, (l, e) => e, 200, true)(() => {
+					setTimeout(() => {
+						this.checkWindowShell();
+					}, 50);
```

Comment:
I think you explained why there was a setTimeout inside the debounce, but I forget why, a comment would be nice